google_guest_agent/diagnostics.go

```diff
@@ -42,15 +42,11 @@ type diagnosticsEntry struct {
 }
 
 func (k diagnosticsEntry) expired() bool {
-	t, err := time.Parse(time.RFC3339, k.ExpireOn)
+	expired, err := utils.CheckExpired(k.ExpireOn)
 	if err != nil {
-		if !containsString(k.ExpireOn, badExpire) {
-			logger.Errorf("Error parsing time: %s", err)
-			badExpire = append(badExpire, k.ExpireOn)
-		}
-		return true
+		logger.Debugf(err.Error())
 	}
-	return t.Before(time.Now())
+	return expired
```

Comment:
in the case where we received an error, what will `expired` be? typically it is undefined, and so you'd want to do a `return true` (treat as expired) in the error handling block

Reply:
Fixed. (Also made the same fix in windows_accounts)